internal/venice-common/src/main/java/com/linkedin/venice/views/MaterializedView.java

```diff
@@ -0,0 +1,57 @@
+package com.linkedin.venice.views;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.meta.ViewParameterKeys;
+import com.linkedin.venice.utils.VeniceProperties;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+
+
+public class MaterializedView extends VeniceView {
+  public static final String MATERIALIZED_VIEW_TOPIC_SUFFIX = "_mv";
+  private static final String MISSING_PARAMETER_MESSAGE = "%s is required for re-partition view!";
+
+  public MaterializedView(Properties props, Store store, Map<String, String> viewParameters) {
+    super(props, store, viewParameters);
+  }
+
+  @Override
+  public Map<String, VeniceProperties> getTopicNamesAndConfigsForVersion(int version) {
+    VeniceProperties properties = new VeniceProperties(props);
+    String viewName = viewParameters.get(ViewParameterKeys.MATERIALIZED_VIEW_NAME.name());
+    return Collections.singletonMap(
+        Version.composeKafkaTopic(store.getName(), version) + VIEW_TOPIC_SEPARATOR + viewName
+            + MATERIALIZED_VIEW_TOPIC_SUFFIX,
+        properties);
+  }
+
+  /**
+   * {@link ViewParameterKeys#MATERIALIZED_VIEW_PARTITION_COUNT} is required to configure a new re-partition view.
+   * {@link ViewParameterKeys#MATERIALIZED_VIEW_PARTITIONER} is optional. The re-partition view will use the store level
+   * partitioner config if it's not specified in the view parameters.
+   * {@link ViewParameterKeys#MATERIALIZED_VIEW_PARTITIONER_PARAMS} is optional.
+   */
+  @Override
+  public void validateConfigs() {
+    super.validateConfigs();
+    String viewName = viewParameters.get(ViewParameterKeys.MATERIALIZED_VIEW_NAME.name());
```

Comment:
Shall we check whether the current store has a view with a same name or not?

Reply:
Good call, let me add that check.